src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseGitDiffReportService.java

```diff
@@ -141,6 +141,23 @@ else if (reports.size() == 1) {
         return latestReport;
     }
 
+    /**
+     * Obtain the {@link ProgrammingExerciseGitDiffReport} of a programming exercise.
+     * If no report exists yet, it will try to generate one
+     *
+     * @param programmingExercise The programming exercise
+     * @return The report or null if none can be generated
+     */
+    public ProgrammingExerciseGitDiffReport getOrCreateReportOfExercise(ProgrammingExercise programmingExercise) {
+        var report = this.getReportOfExercise(programmingExercise);
+        if (report == null) {
+            return updateReport(programmingExercise);
+        }
+        else {
+            return report;
+        }
```

Comment:
```suggestion
        return Objects.requireNonNullElseGet(report, () -> updateReport(programmingExercise));
```

Reply:
I've looked at `requireNonNullElseGet` and it won't work here, as `updateReport` can return null and `requireNonNullElseGet` would throw a NullPointerException in these cases.
`updateReport` returns null when the exercise has not been fully setup yet (e.g. there is no Submission for the template/solution repos yet)

Reply:
Some other code Parts use Optional.ofNullable(...).orElse(...)
Like
```
public Participant getParticipant() {
    return Optional.ofNullable((Participant) student).orElse(team);
}
```